File: server/Models/Entities/Schedule.cs
using System.ComponentModel.DataAnnotations;

namespace server.Models.Entities
{
    public class Schedule
    {
        [Key]
        public Guid ScheduleId { get; set; }

        public string ScheduledBy { get; set; } = string.Empty;

        public Guid? GroupId { get; set; }
        public ResearchGroup? ResearchGroup { get; set; } 


        public DateOnly Date { get; set; }

        public TimeOnly StartTime { get; set; }

        public TimeOnly EndingTime { get; set; }

        [Required]
        public string? RoomVenue { get; set; } 

        public string? AdditionalInformation { get; set; }

        public ICollection<PanelistSchedule> Panelists { get; set; } = [];


   
    }
}

File: packages/types/src/schedule.ts
import type { PanelistScheduleResponseDto } from "./panelist-schedule";
import type { ResearchGroupResponseDto } from "./research-group";

export interface CreatePanelistEntryDto {
  panelistId: string;
  panelistType: string;
}

export interface CreateScheduleDto {
  scheduledBy: string;
  groupId?: string;
  date: string;
  startTime: string;
  endingTime: string;
  roomVenue: string;
  additionalInformation?: string;
  panelists: CreatePanelistEntryDto[];
}

export interface UpdateScheduleDto {
  groupId?: string;
  date?: string;
  startTime?: string;
  endingTime?: string;
  roomVenue?: string;
  additionalInformation?: string;
  panelists?: PanelistScheduleResponseDto[];
}

export interface ScheduleResponseDto {
  scheduleId: string;
  scheduledBy: string;
  researchGroup?: ResearchGroupResponseDto;
  date: string;
  startTime: string;
  endingTime: string;
  roomVenue: string;
  additionalInformation?: string;
  panelists: PanelistScheduleResponseDto[];
}
